marshmallow_utils/fields/localizededtfdatestring.py

```diff
@@ -0,0 +1,135 @@
+
+    def __init__(self, format=None, locale=None):
+        """EDTF formatter consrtuctor.
+
+        :params format: One of `AVAILABLE_FORMATS`.
+        :params locale: Locale value used for localization.
+        """
+        self.format = format or 'medium'
+        self.locale = locale or 'en'
+
+    def has_day_precision(self, edtfdate):
+        """Check if EDTF date has `day` precision.
+
+        If so, then the string can be converted to a python date object.
+        """
+        return edtfdate.precision == 'day'
+
+    def to_date(self, edtfdate, strict="lower"):
+        """Convert EDTF date to python date object.
+        :params strict: Convert EDTF date string to `lower` or `upper` strict
+                        in case it is not a `python.datetime` compatible value.
+        """
+        if self.has_day_precision(edtfdate):
+            return datetime.strptime(edtfdate.isoformat(), "%Y-%m-%d").date()
+        else:
+            if strict == 'lower':
+                edtfdate = edtfdate.lower_strict()
+            elif strict == 'upper':
+                edtfdate = edtfdate.upper_strict()
+            return date.fromtimestamp(calendar.timegm(edtfdate))
+
+    def format_date(self, edtfdate, strict):
+        """Format single EDTF date."""
+        try:
+            # TODO: remove when moved to external package
+            from flask_babelex import format_date
+            date_obj = self.to_date(edtfdate, strict)
+            return format_date(date_obj, self.format, self.locale)
+        except ImportError:
+            return str(edtfdate)
+
+    def format_range_date(self, range_dates):
+        """Format range of EDTF dates."""
+        # TODO: remove when moved to external package
+        try:
+            from flask_babelex import format_date
+            start_date = self.format_date(range_dates[0], strict="lower")
+            end_date = self.format_date(range_dates[1], strict="upper")
+            return self.RANGE_SEPARATOR.join([start_date, end_date])
+        except ImportError:
+            return self.RANGE_SEPARATOR.join(range_dates)
+
+
+class LocalizedEDTFDateString(fields.Str):
+    """
+    Localized Extended Date(/Time) Format Level 0 date string field.
+
+    Made a field for stronger semantics than just a validator.
+    """
+
+    default_error_messages = {
+        "invalid": _("Please provide a valid date or interval.")
+    }
+
+    def __init__(self, formatter=None, format=None, locale=None, **kwargs):
+        """Contructor."""
+        self.formatter = formatter or LocalizedEDTFFormatter(
+            format=format, locale=locale)
+        super(LocalizedEDTFDateString, self).__init__(**kwargs)
+
+    def _parse_date_string(self, datestring):
+        """Parse input string as EDTF."""
+        parser = level0Expression("level0")
+        try:
+            result = parser.parseString(datestring)
+            if not result:
+                raise ParseException()
```

Comment:
We realized in https://github.com/inveniosoftware/marshmallow-utils/pull/16/files#diff-17447330733ca2611754695af731137cf1dcf855e5b8386fd4a5df152f453921L45 that this was not needed. Maybe you have a case, we didn't think of?